express-api/src/services/properties/propertiesServices.ts

```diff
@@ -92,7 +92,14 @@ const sortKeyMapping = (
 
 const collectFindOptions = (filter: PropertyUnionFilter) => {
   const options = [];
-  if (filter.agency) options.push(constructFindOptionFromQuery('Agency', filter.agency));
+  // Has to handle if agency searched by name or set by users agencies
+  if (filter.agency || filter.agencyId)
+    options.push({
+      ...constructFindOptionFromQuery('Agency', filter.agency),
+      AgencyId: filter.agencyId
+        ? In(typeof filter.agencyId === 'number' ? [filter.agencyId] : filter.agencyId)
+        : undefined,
+    });
```

Comment:
I'm having second thoughts about this solution.
This still only contributes to the OR query. If a user has agencyID # 1 only, but they also search by pid, I think they would be able to get a PID outside of their agency...

Reply:
Fixed with recent commits.